lisp/mastodon-tl.el

```diff
@@ -399,21 +399,35 @@ Move forward (down) the timeline unless BACKWARD is non-nil."
   (goto-char (point-max))
   (mastodon-tl--property 'toot-id t))
 
+(defun mastodon-tl--as-string(id)
+  (cond ((numberp id )
+         (number-to-string id))
+        ((stringp id)
+         id)
+        (t (error
+            "id:%s must be either a string or a number"
+            id))))
+
 (defun mastodon-tl--thread ()
   "Open thread buffer for toot under `point'."
   (interactive)
-  (let* ((id (number-to-string (mastodon-tl--property 'toot-id)))
+  (let* ((id (mastodon-tl--as-string (mastodon-tl--property 'toot-id)))
          (url (mastodon-http--api (format "statuses/%s/context" id)))
          (buffer (format "*mastodon-thread-%s*" id))
          (toot (mastodon-tl--property 'toot-json))
          (context (mastodon-http--get-json url)))
     (with-output-to-temp-buffer buffer
       (switch-to-buffer buffer)
+      (mastodon-mode)
+      (setq mastodon-tl--buffer-spec
```

Comment:
Nit: This change seems unrelated to what the commit message says...

Reply:
Updated the commit message to include a line addressing this move. I was having odd behaviour in the thread buffer if i defined mastodon-tl--buffer-spec` before initializing `mastodon-mode`.